_posts/2020-02-20-whats-coming-in-eslint-7.0.0.md

```diff
@@ -0,0 +1,105 @@
+
+Several approaches were considered before we decided to replace the `CLIEngine` class with the new `ESLint` class ([more details](https://github.com/eslint/rfcs/pull/40)). The `ESLint` class will contain asynchronous methods that will serve as a jumping off point to introducing asynchronous functionality into the core. To begin, the `ESLint` class will just be a wrapper around `CLIEngine` and we will systematically start writing new functionality into the `ESLint` class over time. Eventually, the `ESLint` and `CLIEngine` classes will be completely separate.
+
+For v7.0.0, the `CLIEngine` class will be deprecated but not removed. We expect to remove `CLIEngine` in a future major release (don't worry, we'll give you plenty of notice before doing that).
+
+### Deprecating Node.js/CommonJS-specific rules
+
+Managing rules in the core is a significant part of ongoing ESLint maintenance. As the project has grown, so has the number of rules we've maintained. We reviewed all of the current rules and decided that the Node.js/CommonJS-specific rules aren't something that makes sense to include in the core going forward. This is especially true as Node.js moves more towards ESM as the preferred module system.
+
+As such, the following rules will be deprecated in the core:
+
+* `callback-return`
+* `global-require`
+* `handle-callback-err`
+* `no-buffer-constructor`
+* `no-mixed-requires`
+* `no-new-require`
+* `no-path-concat`
+* `no-process-env`
+* `no-process-exit`
+* `no-restricted-modules`
+
+These rules will be moved to the [`eslint-plugin-node`](https://github.com/mysticatea/eslint-plugin-node) project. We will be recommending that people use the plugin for these rules going forward. The rules will remain in ESLint for v7.0.0 and will be removed in a future major release.
+
+### Descriptions in directive comments
+
+Directive comment descriptions allow you to leave a note to yourself and other developers as to why you included the comment. For example:
+
+```js
+/* eslint-disable no-new -- this class has a side-effect in the constructor. */
+```
+
+You can read more in the [RFC](https://github.com/eslint/rfcs/pull/33).
+
+### Deprecating `~/.eslintrc`
+
+Config files in your home directory are deprecated in v7.0.0. This feature has long been a source of confusion for ESLint users, especially because ESLint only looks for a config file in the home directory if no other config file is found. The configuration system in ESLint has changed a lot since home directory config files were introduced, and with the additional flexibility we feel there are plenty of other options for configuring ESLint that are clearer than the home directory config file.
```

Comment:
This deprecation happened in 6.7.0. We add runtime warnings in 7.0.0 for a few breaking cases on removal.

Reply:
Ooh thanks, I'll update that.